File: fastkart_frontend/src/app/shared/models/Product.ts
import { ProductBid } from "./ProductBid";

export class Product{
    "productId": number;
    "productName": String;
    "productCategory": String;
    "productDescription": String;
    "sellerName": String;
    "minBid": number;
    "listedDateTime": Date;
    "minBidByBuyer": number;
    "maxBidByBuyer": number;
    "productBids": ProductBid[];
}

File: FastKart_BE/fastkart-buyer-service/src/main/java/com/fastkart/buyer/entity/Product.java
package com.fastkart.buyer.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;


@Entity
@Table(name="products")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Product {

    @Id
    @Column(name = "product_id")
    private int productId;

    @Column(name = "product_name")
    private String productName;

    @Column(name = "product_category")
    private String productCategory;

    @Column(name = "product_description")
    private String productDescription;

    @Column(name = "min_bid")
    private int minBid;

    @Column(name = "seller_name")
    private String sellerName;

    @Column(name = "listed_date_time")
    private LocalDateTime listedDateTime;

    @Transient
    private int maxBidByBuyer;

    @OneToMany(mappedBy = "product", fetch = FetchType.LAZY,
            cascade = CascadeType.ALL)
    private List<ProductBid> productBids = new ArrayList<>();
}
